src/models/iceflow/SIA2D/SIA2D.jl

```diff
@@ -310,3 +316,26 @@ function initialize_iceflow_model(
     iceflow_model.MB_total = zeros(F,nx,ny)
 end
 
+function build_callback(model::SIA2Dmodel, simulation, glacier_idx, θ)
+    function A_affect!(integrator)
```

Comment:
I remember there was some penalty in declaring functions inside functions, since they act as closures. Could this work by declaring all these affects outside?